### PR TITLE
Archimedes hard fork: Disable sha2, ripemd, blake2f precompiles

### PR DESCRIPTION
**Pull request by maxwolff**
named archimedes 

support an earlier testnet release by pushing back implementation of some precompiles

tested locally as well

### CODE REVIEW

**Comment by maxwolff** [comment]
`eth/tracers/internal/tracetest/calltrace_test.go` fails on develop, apparently bc selfdestruct was deprecated

makes it hard to know if this change would also break that test. just fyi

**Comment by maxwolff** [comment]
did not address making changes in `cmd/puppeth/genesis.go` to support building an aleth config, as aleth is deprecated and we do not use it

File: genesis_zktrie.json [hunk 1]
```diff
@@ -12,6 +12,7 @@
     "istanbulBlock": 0,
     "berlinBlock": 0,
     "londonBlock": 0,
+    "placeholderBlock": 0,
```
Comment (maxwolff) [inline genesis_zktrie.json]: we'll override this file with the genesis file in testnet, i assume

**Review by georgehao — DISMISSED** [review]

**Review by zimpha — DISMISSED** [review]

File: params/config.go
Comment (Thegaram) [inline params/config.go:1]: Also need to update `ScrollAlphaChainConfig` (the built-in default genesis config)

**Review by Thegaram — COMMENTED** [review]
We decided to only disable `SHA2-256`, `RIPEMD-160`, and `blake2f`. Otherwise legit.

File: params/config.go
Comment (maxwolff) [reply params/config.go:1]: with what value? 0? or the block we actually want to fork at?

File: params/config.go [hunk 2]
```diff
@@ -275,6 +275,7 @@ var (
 		BerlinBlock:         big.NewInt(0),
 		LondonBlock:         big.NewInt(0),
 		ArrowGlacierBlock:   nil,
+		ArchimedesBlock:     big.NewInt(0),
```
Comment (Thegaram) [inline params/config.go]: Let's set it to `nil` now and set to a specific block number in another PR.

File: eth/tracers/js/tracer_test.go [hunk 3]
```diff
@@ -225,6 +226,37 @@ func TestIsPrecompile(t *testing.T) {
...
+	blockCtx = vm.BlockContext{BlockNumber: big.NewInt(450)}
+	res, err = runTrace(tracer, &vmContext{blockCtx, txCtx}, chaincfg)
+	if err != nil {
+		t.Error(err)
+	}
+	if string(res) != "false" {
+		t.Errorf("Tracer should not consider blake2f as precompile in scroll alpha")
+	}
+
+	// test blake2f disabled in archimedes
```
Comment (Thegaram) [inline eth/tracers/js/tracer_test.go:251]: This is the same as above?

**Review by Thegaram — DISMISSED** [review]

File: params/config.go
Comment (Thegaram) [reply params/config.go:1]: We can use `nil` for now for backward compatibility.

**Review by zimpha — APPROVED** [review]

**Review by Thegaram — APPROVED** [review]

**Pull request merged**